### ISSUE TITLE
does not create a new line and consumes a lot of CPU resources

### ISSUE THREAD

**Issue by intern** [ohmyzsh/ohmyzsh#12360]
### Describe the bug

Local mac os system, using iterm2 to ssh connect to remote centos 7 system, ohmyzsh latest version installed on centos 7, encountering issues where pressing ctrl c does not create a new line, and zsh consumes a lot of CPU resources after the connection is disconnected.

### Steps to reproduce

Local mac os system, using iterm2 to ssh connect to remote centos 7 system, ohmyzsh latest version installed on centos 7, encountering issues where pressing ctrl c does not create a new line, and zsh consumes a lot of CPU resources after the connection is disconnected.

### Expected behavior

Rolling back to a previous version(15ffbdd) resolves the issue, and I haven't tested the versions in between. I'm not sure which version's changes are causing the problem.

### OS / Linux distribution

Mac os 13.52

### Zsh version

5.9

### Terminal emulator

iTerm2

### If using WSL on Windows, which version of WSL

None

### Additional context

Please Fix it!

**Comment by carlosala** [comment]
Could you `git bisect` and discover which commit creates the issue?

**Comment by intern** [comment]
https://github.com/ohmyzsh/ohmyzsh/commit/ec1afe9dd683c36e6384db25fc1e95acbb0cbc7a

**Comment by carlosala** [comment]
@intern could you update to the latest omz master and see if it is fixed for you?

**Comment by liguoqinjim** [comment]
I encountered the same problem. `Ctrl+C` can still send an INT signal, but it cannot create a new line. Press `Ctrl+C`, then press a letter to create a new line.

**Comment by carlosala** [comment]
Hi @liguoqinjim! Which version of `ohmyzsh` are you running?

**Comment by liguoqinjim** [comment]
> Hi @liguoqinjim! Which version of `ohmyzsh` are you running? Are you as well on centos 7?
Hello, I tried it out and might have found the issue.I hope this helps you solve the problem.🤪

centos verions:CentOS Linux release 7.6.1810 (Core)
zsh version:zsh 5.0.2 (x86_64-redhat-linux-gnu)

correct commit: f78c6b90fc18e2d0bb119ff549207463ce1d57a5
commit that may have issues: 0c80a063c3ad9513b83589d377c501a8d85d63a6

**Comment by mcornella** [comment]
If commit f78c6b9 fixes the issue then this is already solved in the latest version and this should be closed.

**Comment by mcornella** [comment]
Duplicate of #12331.

**Comment by liguoqinjim** [comment]
> If commit [f78c6b9](https://github.com/ohmyzsh/ohmyzsh/commit/f78c6b90fc18e2d0bb119ff549207463ce1d57a5) fixes the issue then this is already solved in the latest version and this should be closed.

Nope,the latest commit `b1c5315a5fbc33624525e434ddbaec3bb30030e2`still has the issue.

**Comment by mcornella** [comment]
And you can turn the behavior off with the following line in your `.zshrc` file, before Oh My Zsh is sourced?

```sh
zstyle ':omz:alpha:lib:git' async-prompt no
```

**Comment by mcornella** [comment]
Can you make a screen recording of the issue, and also put `functions -t _omz_async_request _omz_async_callback` in your zshrc file at the bottom to get a trace?

**Comment by liguoqinjim** [comment]
> And you can turn the behavior off with the following line in your `.zshrc` file, before Oh My Zsh is sourced?
> 
> ```shell
> zstyle ':omz:alpha:lib:git' async-prompt no
> ```

commit id: "b1c5315a5fbc33624525e434ddbaec3bb30030e2"
```
zstyle ':omz:alpha:lib:git' async-prompt no
source $ZSH/oh-my-zsh.sh
```

With the line, result is correct.

**Comment by nasacj** [comment]
> And you can turn the behavior off with the following line in your `.zshrc` file, before Oh My Zsh is sourced?
> 
> ```shell
> zstyle ':omz:alpha:lib:git' async-prompt no
> ```

This works. Could you please help to fix it? So that with running`omz update` can fix this issue @mcornella

**Comment by carlosala** [comment]
> Can you make a screen recording of the issue, and also put `functions -t _omz_async_request _omz_async_callback` in your zshrc file at the bottom to get a trace?

Could someone send the screen recording? 🙏🏻
Maybe @jsrdzhk?
Thanks!

**Comment by jsrdzhk** [comment]
sorry for late response.I uncomment line `ZSH_THEME="robbyrussell"` in ~/.zshrc but did not reproduce the phemomenon.Here's the output for `source ~/.zshrc` after change the zsh profile:
```
+_omz_async_request:1> local -i 'ret=0'
+_omz_async_request:2> typeset -gA _OMZ_ASYNC_FDS _OMZ_ASYNC_PIDS _OMZ_ASYNC_OUTPUT
+_omz_async_request:5> local handler
+_omz_async_request:6> handler=_omz_git_prompt_info
+_omz_async_request:7> ((  1  ))
+_omz_async_request:9> local 'fd=-1'
+_omz_async_request:10> local 'pid=-1'
+_omz_async_request:13> ((  fd != -1 && pid != -1  ))
+_omz_async_request:33> _OMZ_ASYNC_FDS[$handler]=-1
+_omz_async_request:34> _OMZ_ASYNC_PIDS[$handler]=-1
+_omz_async_request:47> _OMZ_ASYNC_FDS[$handler]=12
+_omz_async_request:52> is-at-least 5.8
+is-at-least:17> emulate -L zsh
+is-at-least:19> local 'IFS=.-' 'min_cnt=0' 'ver_cnt=0' part min_ver version
+is-at-least:21> min_ver=( 5 8 )
+is-at-least:22> version=( 5 0 2 0 )
+is-at-least:24> ((  0 <= 2  ))
+is-at-least:25> [[ '' != <-> ]]
+is-at-least:26> ((  ++ver_cnt > 4  ))
+is-at-least:27> part=5
+is-at-least:25> [[ 5 != <-> ]]
+is-at-least:30> true
+is-at-least:31> ((  ++min_cnt > 2  ))
+is-at-least:32> [[ 5 == <-> ]]
+is-at-least:32> break
+is-at-least:35> ((  part > min_ver[min_cnt]  ))
+is-at-least:36> ((  part < min_ver[min_cnt]  ))
+is-at-least:37> part=''
+is-at-least:24> ((  1 <= 2  ))
+is-at-least:25> [[ '' != <-> ]]
+is-at-least:26> ((  ++ver_cnt > 4  ))
+(anon):0> return 0
+is-at-least:27> part=0
+is-at-least:25> [[ 0 != <-> ]]
+is-at-least:30> true
+is-at-least:31> ((  ++min_cnt > 2  ))
+is-at-least:32> [[ 8 == <-> ]]
+is-at-least:32> break
+_omz_async_request:43> _omz_git_prompt_info
+is-at-least:35> ((  part > min_ver[min_cnt]  ))
+is-at-least:36> ((  part < min_ver[min_cnt]  ))
+is-at-least:36> return 1
+_omz_git_prompt_info:3> __git_prompt_git rev-parse --git-dir
+_omz_async_request:52> true
+_omz_async_request:55> read -u 12 '_OMZ_ASYNC_PIDS[_omz_git_prompt_info]'
+_omz_async_request:58> zle -F 12 _omz_async_callback
+_omz_git_prompt_info:5> return 0
```

By the way,I cleaned ~/.zsh_history before.

**Comment by jsrdzhk** [comment]
```
+_omz_async_request:1> local -i 'ret=0'
+_omz_async_request:2> typeset -gA _OMZ_ASYNC_FDS _OMZ_ASYNC_PIDS _OMZ_ASYNC_OUTPUT
+_omz_async_request:5> local handler
+_omz_async_request:6> handler=_omz_git_prompt_info
+_omz_async_request:7> ((  1  ))
+_omz_async_request:9> local 'fd=12'
+_omz_async_request:10> local 'pid=73310'
+_omz_async_request:13> ((  fd != -1 && pid != -1  ))
+_omz_async_request:16> zle -F 12
+_omz_async_request:20> [[ -o MONITOR ]]
+_omz_async_request:23> kill -TERM -73310
+_omz_async_request:33> _OMZ_ASYNC_FDS[$handler]=-1
+_omz_async_request:34> _OMZ_ASYNC_PIDS[$handler]=-1
+_omz_async_request:47> _OMZ_ASYNC_FDS[$handler]=12
+_omz_async_request:52> is-at-least 5.8
+is-at-least:17> emulate -L zsh
+is-at-least:19> local 'IFS=.-' 'min_cnt=0' 'ver_cnt=0' part min_ver version
+is-at-least:21> min_ver=( 5 8 )
+is-at-least:22> version=( 5 0 2 0 )
+is-at-least:24> ((  0 <= 2  ))
+is-at-least:25> [[ '' != <-> ]]
+is-at-least:26> ((  ++ver_cnt > 4  ))
+is-at-least:27> part=5
+is-at-least:25> [[ 5 != <-> ]]
+is-at-least:30> true
+is-at-least:31> ((  ++min_cnt > 2  ))
+is-at-least:32> [[ 5 == <-> ]]
+is-at-least:32> break
+is-at-least:35> ((  part > min_ver[min_cnt]  ))
+is-at-least:36> ((  part < min_ver[min_cnt]  ))
+(anon):0> return 0
+is-at-least:37> part=''
+is-at-least:24> ((  1 <= 2  ))
+is-at-least:25> [[ '' != <-> ]]
+is-at-least:26> ((  ++ver_cnt > 4  ))
+is-at-least:27> part=0
+is-at-least:25> [[ 0 != <-> ]]
+is-at-least:30> true
+_omz_async_request:43> _omz_git_prompt_info
+is-at-least:31> ((  ++min_cnt > 2  ))
+is-at-least:32> [[ 8 == <-> ]]
+is-at-least:32> break
+_omz_git_prompt_info:3> __git_prompt_git rev-parse --git-dir
+is-at-least:35> ((  part > min_ver[min_cnt]  ))
+is-at-least:36> ((  part < min_ver[min_cnt]  ))
+is-at-least:36> return 1
+_omz_async_request:52> true
+_omz_async_request:55> read -u 12 '_OMZ_ASYNC_PIDS[_omz_git_prompt_info]'
+_omz_async_request:58> zle -F 12 _omz_async_callback
+_omz_git_prompt_info:5> return 0
```
The bug reproduced.Here's the output.

**Comment by liguoqinjim** [comment]
> > Can you make a screen recording of the issue, and also put `functions -t _omz_async_request _omz_async_callback` in your zshrc file at the bottom to get a trace?
> 
> Could someone send the screen recording? 🙏🏻 Maybe @jsrdzhk? Thanks!


https://github.com/ohmyzsh/ohmyzsh/assets/2113954/bf914652-3677-4fca-8718-7da893e0bdb7

**Comment by carlosala** [comment]
Thank you! Could you enable the trace for the screen record? Thanks!

**Comment by mcornella** [comment]
I think this is confirmed to be bad performance due to versions < 5.0.6. We should disable the default again for such versions, which ideally would be short-lived because of the (finally) EOL of CentOS 7.

**Comment by carlosala** [comment]
It's indeed. Let's disable the async by default, lgtm 👍🏻

**Comment by jsrdzhk** [comment]
so how to avoid this problem?

**Comment by mcornella** [comment]
Should be fixed in 59e8e028e179fc126d46254900946953072048e7, async prompt has been turned off for zsh 5.0.5 and older. You can get the changes with `omz update`. Thanks!
